website/server/models/task.js

```diff
@@ -96,7 +96,17 @@ export const TaskSchema = new Schema({
     validate: [v => validator.isUUID(v), 'Invalid uuid for task tags.'],
   }],
   // redness or cost for rewards Required because it must be settable (for rewards)
-  value: { $type: Number, default: 0, required: true },
+  value: {
```

Comment:
Unfortunately adding the validator here will cause existing rewards with a negative to be un-saveable due to the new check that would also fails. This will happen when you try to update the rewards or even when scoring it. Could you instead move the check to only happen when the reward is being update or created? Maybe directly in the controller function

Reply:
I see that in the process of scoring a task you automatically convert the value to a positive one so in that case the validation error should not occur but only when updating the reward and keeping a negative value

Reply:
Yes, that's why I've made a change that flips negative reward value into a positive with same number before save. Maybe it would be even better to create a migration script which changes all user rewards with negative values into rewards with positive value?

Reply:
Oh I see, I'll check that behavior and update commit when I got home

Reply:
At this point I think a migration might be the best way to solve this, what do you think @Alys?

Reply:
Yep, I actually wanted to propose this :)

Reply:
I agree that a migration would be good!

Reply:
let's go with the migration then! @hamboomger you can see an example of how to write a migration here https://github.com/HabitRPG/habitica/blob/develop/migrations/users/bulk-email.js:

- a new file with `date_name.js` would go into `/migrations`
- you can search for all users or a part of them tweaking the query here https://github.com/HabitRPG/habitica/blob/develop/migrations/users/bulk-email.js#L29 (the only part that has to stay is the `migration: { $ne: MIGRATION_NAME }`
- and finally update the users here https://github.com/HabitRPG/habitica/blob/develop/migrations/users/bulk-email.js#L14

Let me know if you have any question or need help with the migration

Reply:
@paglias I actually ran into trouble which is described in the related issue, could you take a look at it pls?

Reply:
sorry again for the delay, but I answered there, if you can't find a solution don't worry and we'll handle the migration